src/System.Text.Json/src/System/Text/Json/Writer/Utf8JsonWriter.cs

```diff
@@ -284,13 +284,16 @@ public void Flush()
         {
             CheckNotDisposed();
 
+            _memory = default;
+
             if (_stream != null)
             {
                 Debug.Assert(_arrayBufferWriter != null);
                 if (BytesPending != 0)
                 {
                     _arrayBufferWriter.Advance(BytesPending);
-                    Debug.Assert(BytesPending == _arrayBufferWriter.WrittenCount);
```

Comment:
Was this Debug.Assert invalid?

Reply:
Hmm... I see why this would be invalid. If `Grow` doesn't write to the stream and clear the `_arrayBufferWriter`, then `BytesPending != _arrayBufferWriter.WrittenCount` (from https://github.com/dotnet/corefx/issues/39837)

Reply:
You can see the failures easily if you just run the tests from this PR without the associated fix; they all assert.